api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryServiceImpl.java

```diff
@@ -0,0 +1,294 @@
+                .rpOtherPurpose(rs.getBoolean("rp_other_purpose"))
+                .rpOtherPurposeDetails(rs.getString("rp_other_purpose_details"))
+                .rpPopulationHealth(rs.getBoolean("rp_population_health"))
+                .rpReasonForAllOfUs(rs.getString("rp_reason_for_all_of_us"))
+                .rpReviewRequested(rs.getBoolean("rp_review_requested"))
+                .rpScientificApproach(rs.getString("rp_scientific_approach"))
+                .rpSocialBehavioral(rs.getBoolean("rp_social_behavioral"))
+                .rpTimeRequested(offsetDateTimeUtc(rs.getTimestamp("rp_time_requested")))
+                .workspaceId(rs.getLong("workspace_id")));
+  }
+
+  @Override
+  public List<ReportingCohort> getCohorts() {
+    return jdbcTemplate.query(
+        "SELECT \n"
+            + "  cohort_id,\n"
+            + "  creation_time,\n"
+            + "  creator_id,\n"
+            + "  criteria,\n"
+            + "  description,\n"
+            + "  last_modified_time,\n"
+            + "  name,\n"
+            + "  workspace_id\n"
+            + "FROM cohort",
+        (rs, unused) ->
+            new ReportingCohort()
+                .cohortId(rs.getLong("cohort_id"))
+                .creationTime(offsetDateTimeUtc(rs.getTimestamp("creation_time")))
+                .creatorId(rs.getLong("creator_id"))
+                .criteria(rs.getString("criteria"))
+                .description(rs.getString("description"))
+                .lastModifiedTime(offsetDateTimeUtc(rs.getTimestamp("last_modified_time")))
+                .name(rs.getString("name"))
+                .workspaceId(rs.getLong("workspace_id")));
+  }
+
+  @Override
+  public List<ReportingDataset> getDatasets() {
+    return jdbcTemplate.query(
+        "SELECT \n"
+            + "  creation_time,\n"
+            + "  creator_id,\n"
+            + "  data_set_id,\n"
+            + "  description,\n"
+            + "  includes_all_participants,\n"
+            + "  last_modified_time,\n"
+            + "  name,\n"
+            + "  workspace_id\n"
+            + "FROM data_set",
+        (rs, unused) ->
+            new ReportingDataset()
+                .creationTime(offsetDateTimeUtc(rs.getTimestamp("creation_time")))
+                .creatorId(rs.getLong("creator_id"))
+                .datasetId(rs.getLong("data_set_id"))
+                .description(rs.getString("description"))
+                .includesAllParticipants(rs.getBoolean("includes_all_participants"))
+                .lastModifiedTime(offsetDateTimeUtc(rs.getTimestamp("last_modified_time")))
+                .name(rs.getString("name"))
+                .workspaceId(rs.getLong("workspace_id")));
+  }
+
+  @Override
+  public List<ReportingDatasetCohort> getDatasetCohorts() {
+    return jdbcTemplate.query(
+        "SELECT data_set_id, cohort_id\n"
+            + "FROM data_set_cohort\n"
+            + "ORDER BY data_set_id, cohort_id;",
```

Comment:
Why do these need to be ordered, and the others not? Comment?

Reply:
they don't; good catch.